embedded/ahtree/ahtree.go

```diff
@@ -37,7 +38,7 @@ var ErrInvalidOptions = fmt.Errorf("%w: invalid options", ErrIllegalArguments)
 var ErrorPathIsNotADirectory = errors.New("ahtree: path is not a directory")
 var ErrorCorruptedData = errors.New("ahtree: data log is corrupted")
 var ErrorCorruptedDigests = errors.New("ahtree: hash log is corrupted")
-var ErrAlreadyClosed = errors.New("ahtree: already closed")
```

Comment:
We may want to split this PR into two, one for handling the error code changes, and the other for the test suite changes